src/wutil.cpp

```diff
@@ -711,14 +727,14 @@ unsigned long long fish_wcstoull(const wchar_t *str, const wchar_t **endptr, int
 /// Like wcstod(), but wcstod() is enormously expensive on some platforms so this tries to have a
 /// fast path.
 double fish_wcstod(const wchar_t *str, wchar_t **endptr) {
+    // We can ignore the locale because we use LC_NUMERIC=C!
     // The "fast path." If we're all ASCII and we fit inline, use strtod().
     char narrow[128];
     size_t len = std::wcslen(str);
     size_t len_plus_0 = 1 + len;
-    auto is_digit = [](wchar_t c) { return '0' <= c && c <= '9'; };
-    if (len_plus_0 <= sizeof narrow && std::all_of(str, str + len, is_digit)) {
+    auto is_ascii = [](wchar_t c) { return 0 <= c && c <= 127; };
+    if (len_plus_0 <= sizeof narrow && std::all_of(str, str + len, is_ascii)) {
```

Comment:
Note: I am not sure if the fast path actually helps anymore if we always use LC_NUMERIC=C.

I would appreciate benchmarks on this.